sources/api/netdog/src/main.rs

```diff
@@ -257,10 +257,13 @@ fn remove(args: RemoveArgs) -> Result<()> {
 
 /// Return the current IP address as JSON (intended for use as a settings generator)
 fn node_ip() -> Result<()> {
-    let ip =
+    let ip_string =
         fs::read_to_string(CURRENT_IP).context(error::CurrentIpReadFailed { path: CURRENT_IP })?;
+    // Validate that we read a proper IP address
+    let _valid_ip = IpAddr::from_str(&ip_string).context(error::IpFromString { ip: &ip_string })?;
```

Comment:
Do you need to store it? I've done things like `my_method.context(error::SomeError)?` without having to store the value of `my_method` and without `cargo fmt` complaining about it

Reply:
I don't have to store it, but I thought it made it a bit clearer to the reader that we're explicitly ignoring the IP type that gets returned.

Reply:
`let _ = ...` seems a little more idiomatic for that case

Reply:
> `let _ = ...` seems a little more idiomatic for that case

I've seen it both ways, but I agree `let _ = ` indicates "this returns something that we don't care about" clearly.

Reply:
I changed it to `let _ =`